### PR TITLE
Add params.env file for opendatahub-operator install support, fixes #37

### LINKED ISSUE #37: Add support for params.env file for configurable operator properties
**Is your feature request related to a problem? Please describe.**
Opendatahub Operator expects a params.env file to contain configurable properties for an Open Data Hub component operator.  These properties are typically used to set image names for operator and services. 

**Describe the solution you'd like**
Configurable operator properties should be defined in a params.env file that can be modified by opendatahub-operator. It should be used to make image names for operator and grpc and rest services configurable. 

**Describe alternatives you've considered**
None. 

**Additional context**
None.

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description

Add a params.env file and config map generator to load the properties, and kustomize replacement to inject the properties in operator deployment. 

## How Has This Been Tested?

Tested by deploying locally and verifying config changes are copied from params.env properties. 

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lampajr — DISMISSED** [review]

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Adds namespace to all resources.
-namespace: model-registry-operator-system
+namespace: opendatahub
```
Comment (tarilabs) [inline config/default/kustomization.yaml:2]: I note the operator will move into the ODH namespace operator

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -70,7 +70,7 @@ spec:
         - /manager
         args:
         - --leader-elect
-        image: quay.io/opendatahub/model-registry-operator:latest
+        image: controller
```
Comment (tarilabs) [inline config/manager/manager.yaml:73]: I'm wondering however if the images are placed in env file, what is the strategy for the `config/manager/kustomization.yaml` file, please? 
I mean wouldn't that operate some changes which now should pertain to the env file?
(open question)

**Review by tarilabs — DISMISSED** [review]
one question; beside that, thank you looks good to me!

**Comment by tarilabs** [comment]
If possible, I'd hold to merge this _after_ Wednesday, December 6 h17:00 UTC so to give time for the demo to finalize using the current quay image.

File: config/default/kustomization.yaml (hunk 1 above)
Comment (dhirajsb) [reply config/default/kustomization.yaml:2]: Yes, we have to align it with other odh components, which all use the same namespace. 
But, I'll investigate whether we can keep the current kubebuilder default config and use an overlay instead for odh. That way the operator can be used without having to do things the _odh_ way.

File: config/manager/manager.yaml (hunk 2 above)
Comment (dhirajsb) [reply config/manager/manager.yaml:73]: Moved odh config to a separate overlay so the default kustomize image override will still work.

**Comment by dhirajsb** [comment]
It's ok to merge this now since it's an entirely separate overlay and doesn't affect default operator deployment.

**Pull request merged**